src/js/components/TextInput/stories/typescript/Custom.tsx

```diff
@@ -0,0 +1,178 @@
+    imageUrl:
+      'https://s.gravatar.com/avatar/4ec9c3a91da89f278e4482811caad7f3?s=80',
+  },
+];
+
+const CustomSuggestionsTextInput = () => {
+  const [value, setValue] = useState('');
+  const [suggestionOpen, setSuggestionOpen] = useState(false);
+  const [suggestedFolks, setSuggestedFolks] = useState([]);
+
+  const [, updateState] = useState();
+  const forceUpdate = useCallback(() => updateState({}), []);
+
+  const boxRef = useRef();
+
+  useEffect(() => {
+    forceUpdate();
+  }, [forceUpdate]);
+
+  const onChange = event => {
+    const { value: newValue } = event.target;
+    setValue(newValue);
+
+    if (!newValue.trim()) {
+      setSuggestedFolks([]);
+    } else {
+      // simulate an async call to the backend
+      setTimeout(() => setSuggestedFolks(folks), 300);
+    }
+  };
+
+  const onSelect = event => setValue(event.suggestion.value);
+
+  const renderSuggestions = () => {
+    return suggestedFolks
+      .filter(
+        ({ name }) => name.toLowerCase().indexOf(value.toLowerCase()) >= 0,
+      )
+      .map(({ name, imageUrl }, index, list) => ({
+        label: (
+          <Box
+            direction="row"
+            align="center"
+            gap="small"
+            border={index < list.length - 1 ? 'bottom' : undefined}
+            pad="small"
+          >
+            <Image
+              width="48px"
+              src={imageUrl}
+              style={{ borderRadius: '100%' }}
+            />
+            <Text>
+              <strong>{name}</strong>
+            </Text>
+          </Box>
+        ),
+        value: name,
+      }));
+  };
+
+  return (
+    <Grommet theme={myCustomTheme} full>
+      <Box background="dark-1" fill align="center" pad={{ top: 'large' }}>
+        <Box
+          ref={boxRef}
+          width="large"
+          direction="row"
+          align="center"
+          pad={{ horizontal: 'small', vertical: 'xsmall' }}
+          round="small"
+          elevation={suggestionOpen ? 'medium' : undefined}
+          border={{
+            side: 'all',
+            color: suggestionOpen ? 'transparent' : 'border',
+          }}
+          style={
+            suggestionOpen
+              ? {
+                  borderBottomLeftRadius: '0px',
+                  borderBottomRightRadius: '0px',
+                }
+              : undefined
+          }
+        >
+          <Search color="brand" />
+          <TextInput
+            type="search"
+            dropTarget={boxRef.current}
+            plain
+            value={value}
+            onChange={onChange}
+            onSelect={onSelect}
+            suggestions={renderSuggestions()}
+            placeholder="Enter your name..."
+            onSuggestionsOpen={() => setSuggestionOpen(true)}
+            onSuggestionsClose={() => setSuggestionOpen(false)}
+          />
+        </Box>
+      </Box>
+    </Grommet>
+  );
+};
+
+storiesOf('TextInput', module).add('Custom', () => (
+  <CustomSuggestionsTextInput />
+));
```

Comment:
Can you please take a look at how other tsx files are taking care of publishing to storybook?
They include the isChromatic query and place the story under the TypeScript directory.

Reply:
I pushed a fix for it.